python/cudf/cudf/core/dataframe.py

```diff
@@ -4078,6 +4084,44 @@ def apply(
         1     4.8
         2     5.0
         dtype: float64
+
+        String data is allowed, so long as its use does not modify or create
+        a new string. The following UDF is allowed:
+
+        >>> def f(row):
+        ...     st = row['str_col']
+        ...     scale = row['scale']
+        ...     if len(st) == 0:
+        ...             return -1
+        ...     elif st.startswith('a'):
+        ...             return 1 - scale
+        ...     elif 'example' in st:
+        ...             return 1 + scale
+        ...     else:
+        ...             return 42
+        ...
+        >>> df = cudf.DataFrame({
+        ...     'str_col': ['', 'abc', 'some_example'],
+        ...     'scale': [1, 2, 3]
+        ... })
+        >>> df.apply(f, axis=1)
+        0   -1
+        1   -1
+        2    4
+        dtype: int64
+
+        However the following UDF is not allowed, because it includes an
+        operation that requires the creation of a new string, namely the
+        ``upper`` method. Methods that are not supported in this manner
+        will raise an ``AttributeError``.
```

Comment:
```suggestion
        However, the following UDF is not allowed since it includes an
        operation that requires the creation of a new string: a call to the
        ``upper`` method. Methods that are not supported in this manner
        will raise an ``AttributeError``.
```